libcontainer/factory_linux.go

```diff
@@ -50,28 +50,51 @@ func InitArgs(args ...string) func(*LinuxFactory) error {
 	}
 }
 
+func getUnifiedPath(paths map[string]string) string {
+	path := ""
+	for k, v := range paths {
+		if path == "" {
+			path = v
+		} else if v != path {
+			panic(errors.Errorf("expected %q path to be unified path %q, got %q", k, path, v))
+		}
+	}
+	// can be empty
+	if path != "" {
+		if filepath.Clean(path) != path || !filepath.IsAbs(path) {
+			panic(errors.Errorf("invalid dir path %q", path))
+		}
+	}
+
+	return path
+}
+
+func systemdCgroupV2(l *LinuxFactory, rootless bool) error {
+	l.NewCgroupsManager = func(config *configs.Cgroup, paths map[string]string) cgroups.Manager {
+		return systemd.NewUnifiedManager(config, getUnifiedPath(paths), rootless)
```

Comment:
Can be another PR, but eventually we should remove `getUnifiedPath()` madness

Reply:
Yes, totally agree, I plan to do that, but this PR is way too big already, let's do it step by step and one day we'll get there :)